csrc/python_frontend/fusion_record.h

```diff
@@ -611,6 +611,93 @@ struct PermuteOpRecord : RecordFunctor {
   std::vector<int64_t> dims_;
 };
 
+// TODO: should this one be merged with PermuteOpRecord?
+struct SetStrideOrderOpRecord : RecordFunctor {
+  SetStrideOrderOpRecord(
+      std::vector<State> _args,
+      std::vector<State> _outputs,
+      std::vector<int64_t> stride_order)
+      : RecordFunctor(
+            std::move(_args),
+            std::move(_outputs),
+            "ops.set_stride_order",
+            serde::RecordType_SetStrideOrderOp),
+        stride_order_(std::move(stride_order)) {}
+  ~SetStrideOrderOpRecord() override = default;
+  RecordFunctor* clone() final {
+    return new SetStrideOrderOpRecord(*this);
+  }
+
+  size_t hash() const final {
+    auto result = RecordFunctor::hash();
+    size_t dims_hash = 0;
+    for (auto dim : stride_order_) {
+      dims_hash ^= static_cast<size_t>(dim);
```

Comment:
This is permutation-invariant I think, so will have same hash for any inputs with same dimension. Maybe just shift by position before xoring, or use `hashCombine`?

Reply:
ohohoh. good catch! We need to fix this for permute as well. :laughing: